src/test/java/com/datatorrent/netlet/AbstractClientTest.java

```diff
@@ -127,16 +127,22 @@ public void sendData()
   {
   }
 
-  @SuppressWarnings("SleepWhileInLoop")
-  private void verifySendReceive(final DefaultEventLoop el, final int port) throws IOException, InterruptedException
+  @SuppressWarnings( {"SleepWhileInLoop", "AssertEqualsBetweenInconvertibleTypes"})
+  private void verifySendReceive(final DefaultEventLoop el) throws IOException, InterruptedException
   {
     ServerImpl si = new ServerImpl();
     ClientImpl ci = new ClientImpl();
 
     new Thread(el).start();
 
-    el.start("localhost", port, si);
-    el.connect(new InetSocketAddress("localhost", port), ci);
+    el.start("localhost", 0, si);
+    synchronized (si) {
```

Comment:
You are right. I missed this. Will change it to appropriately.

From: Vlad Rozov <notifications@github.com<mailto:notifications@github.com>>
Reply-To: DataTorrent/Netlet <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Wednesday, January 13, 2016 at 7:20 AM
To: DataTorrent/Netlet <Netlet@noreply.github.com<mailto:Netlet@noreply.github.com>>
Cc: Chetan Narsude <chetan@apache.org<mailto:chetan@apache.org>>
Subject: Re: [Netlet] make the tests pass by binding to available port as opposed to fixed … (#41)


In src/test/java/com/datatorrent/netlet/AbstractClientTest.java<https://github.com/DataTorrent/Netlet/pull/41#discussion_r49602458>:

>    {
>      ServerImpl si = new ServerImpl();
>      ClientImpl ci = new ClientImpl();
>
>      new Thread(el).start();
>
> -    el.start("localhost", port, si);
> -    el.connect(new InetSocketAddress("localhost", port), ci);
> +    el.start("localhost", 0, si);
> +    synchronized (si) {


while or if(si.getServerAddress() == null) {
  si.wait();
}


—
Reply to this email directly or view it on GitHub<https://github.com/DataTorrent/Netlet/pull/41/files#r49602458>.